### PR TITLE
[INLONG-9009][Sort] Add HBase source and sink connector on flink 1.15

### LINKED ISSUE #9009: [Feature][Sort] Add HBase connector on flink 1.15
### Description

Parent issue: https://github.com/apache/inlong/issues/8336
Add HBase connector on flink 1.15

### Use case

Add HBase connector on flink 1.15

### Are you willing to submit PR?

- [X] Yes, I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by fancycoderzf**
### Prepare a Pull Requestt

- Fixes #9009

### Motivation

*Add HBase source and sink connector on flink 1.15*

### Modifications

*Add HBase source and sink connector on flink 1.15*

### Verifying this change

*(Please pick either of the following options)*

- [ ] This change is a trivial rework/code cleanup without any test coverage.

- [ ] This change is already covered by existing tests, such as:
  *(please describe tests)*

- [ ] This change added tests and can be verified as follows:

  *(example:)*
  - *Added integration tests for end-to-end deployment with large payloads (10MB)*
  - *Extended integration test for recovery after broker failure*

### Documentation

  - Does this pull request introduce a new feature? (yes / no)
  - If yes, how is the feature documented? (not applicable / docs / JavaDocs / not documented)
  - If a feature is not applicable for documentation, explain why?
  - If a feature is not documented yet in this PR, please create a follow-up issue for adding the documentation

### CODE REVIEW

File: inlong-sort/sort-end-to-end-tests/sort-end-to-end-tests-v1.15/src/test/java/org/apache/inlong/sort/tests/HBaseSourceSinkTest.java [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+import java.net.URISyntaxException;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.time.Duration;
+import java.util.Arrays;
+import java.util.List;
+import java.util.stream.Collectors;
+
+/**
+ * @author zfancy
```
Comment (EMsnap) [inline inlong-sort/sort-end-to-end-tests/sort-end-to-end-tests-v1.15/src/test/java/org/apache/inlong/sort/tests/HBaseSourceSinkTest.java:45]: please remove author info and add description about the class

File: inlong-sort/sort-end-to-end-tests/sort-end-to-end-tests-v1.15/pom.xml [hunk 2]
```diff
@@ -132,6 +132,18 @@
             <artifactId>clickhouse-jdbc</artifactId>
             <scope>test</scope>
         </dependency>
+        <dependency>
+            <groupId>junit</groupId>
+            <artifactId>junit</artifactId>
+            <version>4.13</version>
```
Comment (dockerzhang) [inline inlong-sort/sort-end-to-end-tests/sort-end-to-end-tests-v1.15/pom.xml:138]: reuse `<version>${junit.version}</version>`.

File: inlong-sort/sort-flink/sort-flink-v1.15/sort-connectors/hbase/pom.xml [hunk 3]
```diff
@@ -0,0 +1,244 @@
...
+            <version>1.10.0-SNAPSHOT</version>
+            <scope>test</scope>
+        </dependency>
+        <dependency>
+            <groupId>org.apache.inlong</groupId>
+            <artifactId>sort-core</artifactId>
+            <version>1.10.0-SNAPSHOT</version>
+            <scope>test</scope>
+        </dependency>
+        <!--        only for test-->
```
Comment (dockerzhang) [inline inlong-sort/sort-flink/sort-flink-v1.15/sort-connectors/hbase/pom.xml:81]: remove the annotation code.

**Comment by dockerzhang** [comment]
please fix the failed  workflow and conflicting files.

File: inlong-sort/sort-end-to-end-tests/sort-end-to-end-tests-v1.15/src/test/java/org/apache/inlong/sort/tests/HBaseSourceSinkTest.java (hunk 1 above)
Comment (fancycoderzf) [reply inlong-sort/sort-end-to-end-tests/sort-end-to-end-tests-v1.15/src/test/java/org/apache/inlong/sort/tests/HBaseSourceSinkTest.java]: I have removed this file.

File: inlong-sort/sort-flink/sort-flink-v1.15/sort-connectors/hbase/pom.xml (hunk 3 above)
Comment (fancycoderzf) [reply inlong-sort/sort-flink/sort-flink-v1.15/sort-connectors/hbase/pom.xml]: done

**Pull request closed without merging**